src/Mpociot/Versionable/VersionableTrait.php

```diff
@@ -173,10 +183,10 @@ protected function versionablePostSave()
             $version->versionable_id   = $this->getKey();
             $version->versionable_type = method_exists($this, 'getMorphClass') ? $this->getMorphClass() : get_class($this);
             $version->user_id          = $this->getAuthUserId();
-            
+
             $versionedHiddenFields = $this->versionedHiddenFields ?? [];
             $this->makeVisible($versionedHiddenFields);
-            $version->model_data       = serialize($this->attributesToArray());
+            $version->model_data = $this->getEncoding() ? json_encode($this->attributesToArray()) : serialize($this->attributesToArray());
```

Comment:
Hey! Shouldn't this be `$version->model_data = $this->getEncoding() == 'json' ? json_encode($this->attributesToArray()) : serialize($this->attributesToArray());`?

Reply:
indeed! .. goes to show that this PR needs tests

Reply:
Yes, it does! I was testing manually yesterday and it doesn't seem to quite work with my site.